index.js

```diff
@@ -30,6 +31,27 @@ function deploy(workingDir, baseURL, branchOrPack, config, cb) {
   if (shell.test('-f', path.resolve(branchOrPack))) {
     return httpPutDeployment(baseURL, config, branchOrPack, cb);
   } else {
+    if (!branchOrPack || branchOrPack === 'deploy') {
```

Comment:
Because deploy is the default, its being used as a flag indicating that we are in the "default" state... so at this point the default is being changed from "deploy" to the npm pack file name, if it exists.

But this doesn't work. If you explicitly do `slc deploy http://localhost deploy`, and a ../name-version.tgz file exists, this will override the explicit choice of the user.

I suggest that defaults should be set in one place, at https://github.com/strongloop/strong-deploy/blob/master/bin/sl-deploy.js#L69

That simple default of 'deploy' should be replaced with something like (pseudo-code):
```
isGit = ... check if this is a git repo... there is code here or in strong-build to check this.
branchOrPack = branchOrPack || (isGit ? 'deploy' : defaultPkg())
function defaultPkg() {
  return path.join("..", "<name>-<version>.tgz")
}
```

That way, the default is applied in only one place (not two files), and the default doesn't ever override an explicit CLI option.